src/Features/Core/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -291,7 +291,7 @@ private bool CheckOption(Workspace workspace, string language, bool documentOpen
         {
             try
             {
-                if (!skipClosedFileChecks && !CheckOption(project.Solution.Workspace, project.Language, documentOpened: project.Documents.Any(d => d.IsOpen())))
+                if (!skipClosedFileChecks && !CheckOption(project.Solution.Workspace, project.Language, documentOpened: false))
```

Comment:
ya, that is the intention. we don't want to run those when the flag is off. it should make it easier for v2 to implement this flag as well.

if it does all work and then at the end, filter close file diagnostics, then it defeats reason why the flag is there at the first place.